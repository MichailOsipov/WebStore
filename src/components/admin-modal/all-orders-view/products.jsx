import * as React from 'react';
import {ProductCard} from 'components/product-card';

export const Products = ({products}) => (
    <div>
        {products.map(({id, name, imageSrc}) => (
            <ProductCard
                key={id}
                name={name}
                imageSrc={imageSrc}
            />
        ))}
    </div>
);
