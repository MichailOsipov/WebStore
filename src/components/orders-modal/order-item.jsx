import * as React from 'react';

export const OrderItem = ({products}) => (
    <div>
        {products.map(({id, name}) => (
            <div key={id}>{name}</div>
        ))}
    </div>
);
