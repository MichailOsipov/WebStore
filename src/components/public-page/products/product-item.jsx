import * as React from 'react';
import {Button} from '@material-ui/core';

export const ProductItem = ({id, name, onOpenProductInformation}) => (
    <div key={id}>
        {name}
        <Button variant="contained" onClick={onOpenProductInformation}>Подробнее</Button>
    </div>
);
