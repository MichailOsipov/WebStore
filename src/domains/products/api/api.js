import {MOCK_PRODUCTS} from './mock-products';
import {MOCK_PRODUCT} from './mock-product';
import {MOCK_PRODUCTS_ADDED_TO_CART} from './mock-products-added-to-cart';
import {MOCK_ORDERS} from './mock-orders';

export const ProductsApi = {
    loadProducts: () => Promise.resolve(MOCK_PRODUCTS),
    loadProduct: productId => Promise.resolve(MOCK_PRODUCT),
    addProductToCart: productId => Promise.resolve(),
    loadProductsAddedToCart: () => Promise.resolve(MOCK_PRODUCTS_ADDED_TO_CART),
    removeProductFromCart: productId => Promise.resolve(),
    makeAnOrder: () => Promise.resolve(),
    loadOrders: () => Promise.resolve(MOCK_ORDERS)
};
