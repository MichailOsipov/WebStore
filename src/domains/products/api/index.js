export {ProductsApi} from './api';
export {MOCK_ALL_ORDERS} from './mock-all-orders';
export {MOCK_ORDERS} from './mock-orders';
export {MOCK_PRODUCT} from './mock-product';
export {MOCK_PRODUCTS_ADDED_TO_CART} from './mock-products-added-to-cart';
export {MOCK_PRODUCTS} from './mock-products';
