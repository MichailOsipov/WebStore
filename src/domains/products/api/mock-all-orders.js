export const MOCK_ALL_ORDERS = [
    {
        user: 'Андрей',
        orders: [
            {
                id: 3,
                products: [
                    {
                        id: 4,
                        name: 'Автомобиль',
                        imageSrc: 'https://media.tvzvezda.ru/news/vstrane_i_mire/content/201811191113-bhlk.htm/1.jpg'
                    },
                    {
                        id: 5,
                        name: 'Холодильник',
                        imageSrc: 'https://50.img.avito.st/640x480/3764525250.jpg'
                    }
                ]
            },
            {
                id: 4,
                products: [
                    {
                        id: 6,
                        name: 'Лопата',
                        imageSrc: 'https://zs-rus.ru/wp-content/uploads/2017/08/fcb45dbfbd1acf04a8965f251d76d573-1.jpg'
                    }
                ]
            }
        ]
    },
    {
        user: 'Никита',
        orders: [
            {
                id: 5,
                products: [
                    {
                        id: 7,
                        name: 'Стол',
                        imageSrc: 'https://www.nv-lab.ru/images/upload/editor5036.jpg'
                    }
                ]
            }
        ]
    }
];
