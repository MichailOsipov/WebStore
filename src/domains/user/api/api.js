export const UserApi = {
    registerUser: ({name, login, password}) => Promise.resolve({name, role: 'user'}),
    loginUser: ({login, password}) => Promise.resolve({name: 'Mikhail', role: 'admin'})
};
