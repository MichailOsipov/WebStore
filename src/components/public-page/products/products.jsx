import * as React from 'react';
import {connect} from 'react-redux';
import {Grid, Typography} from '@material-ui/core';
import {getProducts, fetchProducts} from 'domains/products/state';
import {ProductItem} from './product-item';

export const Products = connect(
    state => ({
        products: getProducts(state)
    }),
    {
        fetchProducts
    }
)(
    class extends React.Component {
        componentDidMount() {
            this.props.fetchProducts();
        }

        render() {
            const {onOpenProductInformation, products} = this.props;
            return (
                <Grid container justify="center">
                    <Grid item>
                        <Typography variant="h6">Товары в наличии:</Typography>
                    </Grid>
                    <Grid container justify="center">
                        {products.map(({id, name, imageSrc}) => (
                            <Grid item xs={2} key={id}>
                                <ProductItem
                                    name={name}
                                    imageSrc={imageSrc}
                                    onOpenProductInformation={() => onOpenProductInformation(id)}
                                />
                            </Grid>
                        ))}
                    </Grid>
                </Grid>
            );
        }
    });
