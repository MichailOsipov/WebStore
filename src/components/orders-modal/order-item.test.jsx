import * as React from 'react';
import {shallow} from 'enzyme';
import {MOCK_ORDERS} from 'domains/products/api';
import {OrderItem} from './order-item';

describe('OrderItem', () => {
    test('rendering', () => {
        const wrapper = shallow(
            <OrderItem
                products={MOCK_ORDERS[0].products}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
