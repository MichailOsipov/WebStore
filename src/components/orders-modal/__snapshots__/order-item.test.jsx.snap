// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem rendering 1`] = `
<WithStyles(Grid)
  container={true}
  direction="row"
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
    key="5"
    xs={3}
  >
    <ProductCard
      imageSrc="https://images1.popmeh.ru/upload/img_cache/9e6/9e663df2539446f6c910e19e382dd38f_ce_1838x980x80x258_cropped_800x427.jpg"
      name="Автомобиль"
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    key="6"
    xs={3}
  >
    <ProductCard
      imageSrc="https://50.img.avito.st/640x480/3764525250.jpg"
      name="Холодильник"
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
